src/main/java/cz/student/shaccson/UserDao.java

```diff
@@ -0,0 +1,104 @@
+package cz.student.shaccson;
+
+import org.mindrot.jbcrypt.BCrypt;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserDao {
+    private final Connection conn;
+
```

Comment:
CLEAN-CODE: Vytvorit kontanty pro sql dotazy a nasledne je pouzit v metodach.

```suggestion
  private static final String CREATE_USER_QUERY =
            "INSERT INTO users(username, email, password) VALUES (?, ?, ?)";

    private static final String READ_USER_QUERY =
            "SELECT * FROM users WHERE id = ?";

    private static final String UPDATE_USER_QUERY =
            "UPDATE users SET username = ?, email = ?, password = ? WHERE id = ?";

    private static final String DELETE_USER_QUERY =
            "DELETE FROM users WHERE id = ?";

    private static final String SELECT_ALL_USERS_QUERY =
            "SELECT * FROM users";
```

Reply:
opraveno